services/sharepoint-connector/src/permissions-sync/sync-sharepoint-files-permissions-to-unique.command.ts

```diff
@@ -0,0 +1,181 @@
+        this.extractUniqueFileAccessInputsFromUniqueFile(uniqueFile);
+
+      const permissionsToAdd = differenceWith(
+        uniqueFileAccessInputsFromSharePoint,
+        uniqueFileAccessInputsFromUnique,
+        isDeepEqual,
+      );
+      this.logger.debug(`${loopLogPrefix} ${permissionsToAdd.length} permissions to add`);
+      if (permissionsToAdd.length > 0) {
+        const currentScopePermissionsToAdd = permissionsToAddByScopeId[uniqueFile.ownerId] ?? [];
+        permissionsToAddByScopeId[uniqueFile.ownerId] =
+          currentScopePermissionsToAdd.concat(permissionsToAdd);
+      }
+
+      const permissionsToRemove = differenceWith(
+        uniqueFileAccessInputsFromUnique,
+        uniqueFileAccessInputsFromSharePoint,
+        isDeepEqual,
+      );
+      this.logger.debug(`${loopLogPrefix} ${permissionsToRemove.length} permissions to remove`);
+      if (permissionsToRemove.length > 0) {
+        const currentScopePermissionsToRemove =
+          permissionsToRemoveByScopeId[uniqueFile.ownerId] ?? [];
+        permissionsToRemoveByScopeId[uniqueFile.ownerId] =
+          currentScopePermissionsToRemove.concat(permissionsToRemove);
+      }
+    }
+
+    this.logger.log(
+      `${logPrefix} Adding permissions to unique files in ${Object.keys(permissionsToAddByScopeId).length} scopes`,
+    );
+    for (const [scopeId, permissionsToAdd] of Object.entries(permissionsToAddByScopeId)) {
+      this.logger.debug(
+        `${logPrefix}[Scope: ${scopeId}] Adding ${permissionsToAdd.length} permissions`,
+      );
+      await this.uniqueFilesService.addAccesses(scopeId, permissionsToAdd);
+    }
+
+    this.logger.log(
+      `${logPrefix} Removing permissions from unique files in ${Object.keys(permissionsToRemoveByScopeId).length} scopes`,
+    );
+    for (const [scopeId, permissionsToRemove] of Object.entries(permissionsToRemoveByScopeId)) {
+      this.logger.debug(
+        `${logPrefix}[Scope: ${scopeId}] Removing ${permissionsToRemove.length} permissions`,
+      );
+      await this.uniqueFilesService.removeAccesses(scopeId, permissionsToRemove);
+    }
+  }
+
+  private extractUniqueFileAccessInputsFromUniqueFile(
+    uniqueFile: UniqueFile,
+  ): UniqueFileAccessInput[] {
+    return uniqueFile.fileAccess.map((fileAccessKey) => {
+      const fileAccessKeyMatch = /(u|g):(.+)(R|W|M)/.exec(fileAccessKey);
+      assert.ok(fileAccessKeyMatch, `Invalid file access key: ${fileAccessKey}`);
+      const [, granteeType, entityId, accessModifier] = fileAccessKeyMatch;
+      assert.ok(
+        granteeType && entityId && accessModifier,
+        `Invalid file access key: ${fileAccessKey}`,
+      );
+      return {
+        contentId: uniqueFile.id,
+        accessType:
+          ({ R: 'READ', W: 'WRITE', M: 'MANAGE' } as const)[accessModifier] ??
+          assert.fail(`Invalid access modifier: ${accessModifier}`),
+        entityId,
+        entityType: granteeType === 'u' ? 'USER' : 'GROUP',
```

Comment:
maybe we should lowercase or uppercase if they are not coming back consistent as in future they might make it consistent and our code will break